crates/redis/src/lib.rs

```diff
@@ -2,27 +2,29 @@
 
 mod spin;
 
-use std::collections::HashMap;
-
-use anyhow::{anyhow, Context, Result};
+use anyhow::{anyhow, bail, Context, Result};
 use futures::{future::join_all, StreamExt};
 use redis::{Client, ConnectionLike};
 use serde::{de::IgnoredAny, Deserialize, Serialize};
 use spin_core::async_trait;
 use spin_trigger::{cli::NoArgs, TriggerAppEngine, TriggerExecutor};
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
 
 use crate::spin::SpinRedisExecutor;
 
 pub(crate) type RuntimeData = ();
 pub(crate) type Store = spin_core::Store<RuntimeData>;
 
+type ChannelComponents = HashMap<String, Vec<String>>;
 /// The Spin Redis trigger.
+#[derive(Clone)]
 pub struct RedisTrigger {
-    engine: TriggerAppEngine<Self>,
-    // Redis address to connect to
-    address: String,
-    // Mapping of subscription channels to component IDs
-    channel_components: HashMap<String, Vec<String>>,
+    engine: Arc<TriggerAppEngine<Self>>,
+    // List of Redis address to connect to
+    addresses: Vec<String>,
+    // Mapping of server with subscription channel and associated component IDs
```

Comment:
I believe `address` and `server` refer to the same thing - could this be confusing?  Is there a term we can helpfully unify on?

Reply:
server url maybe?